### PR TITLE
close exclusive streams onPause(), fix Issue 448

### PR DESCRIPTION
**Pull request by atneya-nair**
Refactoring various samples to close exclusive stream onPause() per best practice.

### CODE REVIEW

File: samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java [hunk 1]
```diff
@@ -81,18 +83,34 @@ public boolean onTouchEvent(MotionEvent event) {
...
+        // Return to defaults
+        mChannelCountSpinner.setSelection(1);
+        mPlaybackDeviceSpinner.setSelection(0);
+        mBufferSizeSpinner.setSelection(0);
+        mAudioApiSpinner.setSelection(0);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
```
Comment (philburk) [inline samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java]: It is generally better to tear down in reverse order or startup. So move call to super.onPause() to end of onPause().

File: samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 import android.os.Build;
 import android.os.Bundle;
 import androidx.core.view.MotionEventCompat;
+
+import android.util.Log;
```
Comment (dturner) [inline samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java]: Is this being used?

File: samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java [hunk 3]
```diff
@@ -81,18 +83,34 @@ public boolean onTouchEvent(MotionEvent event) {
...
+    */
+    @Override
+    protected void onResume() {
+        super.onResume();
         PlaybackEngine.create(this);
-
-        // Periodically update the UI with the output stream latency
-        mLatencyText = findViewById(R.id.latencyText);
         setupLatencyUpdater();
+        // Return to defaults
```
Comment (dturner) [inline samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java]: Is this new functionality? If so, please split into a separate pull request.

File: samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java [hunk 4]
```diff
@@ -234,8 +249,6 @@ public void run() {
 
     @Override
     protected void onDestroy() {
```
Comment (dturner) [inline samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java]: This method is only calling the parent `onDestroy` so can safely be removed

File: samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java [hunk 5]
```diff
@@ -81,18 +81,34 @@ public boolean onTouchEvent(MotionEvent event) {
...
+    @Override
+    protected void onResume() {
+        super.onResume();
         PlaybackEngine.create(this);
-
-        // Periodically update the UI with the output stream latency
-        mLatencyText = findViewById(R.id.latencyText);
         setupLatencyUpdater();
+        // Return the spinner states to their default value
+        mChannelCountSpinner.setSelection(1);
```
Comment (dturner) [inline samples/hello-oboe/src/main/java/com/google/sample/oboe/hellooboe/MainActivity.java]: Nit: It'd be clearer to use named constants here e.g. `private final int kDefaultChannelCountSelection = 1`
